src/sentry/tasks/update_user_reports.py

```diff
@@ -19,16 +20,43 @@ def update_user_reports(**kwargs):
         group__isnull=True, environment__isnull=True, date_added__gte=now - timedelta(days=1)
     )
 
-    count = 0
-    updated = 0
-    for report in user_reports:
-        count += 1
-        event = eventstore.get_event_by_id(report.project_id, report.event_id)
-        if event:
-            report.update(group_id=event.group_id, environment=event.get_environment())
-            updated += 1
+    # We do one query per project, just to avoid the small case that two projects have the same event ID
+    project_map = {}
+    for r in user_reports:
+        project_map.setdefault(r.project_id, []).append(r)
+
+    # Logging values
+    total_reports = len(user_reports)
+    reports_with_event = 0
+    updated_reports = 0
+    samples = None
+
+    for project_id, reports in six.iteritems(project_map):
+        event_ids = [r.event_id for r in reports]
+        report_by_event = {r.event_id: r for r in reports}
+        snuba_filter = eventstore.Filter(project_ids=[project_id], event_ids=event_ids)
```

Comment:
Is there a time range we can apply to this query? Doing a query across the full retention window results in clickhouse not being able to use its indexes very well. Having a date condition will make these queries much faster.

Reply:
Yeah probably. I'll add one.